src/control/ToolHandler.h

```diff
@@ -23,7 +23,8 @@
 #include "Tool.h"
 #include "XournalType.h"
 
-enum ToolPointer { current, toolbar, button };
+enum SelectedTool { active, toolbar };
```

Comment:
I'm not really happy with "active". Am I right with assuming this is a tool that was selected from the toolbar but then modified?

Reply:
The logic here is the following

There are two kinds of pointers
* `Button` Pointers: will not change the tool they are pointing to over their lifetime unless the settings are changed
* `SelectedTool` Pointers: will mostly change the tool they are pointing to over their lifetime

The meaning of the button pointers is self explanatory (I hope).

For the Toolpointer it goes as follows:
* `toolbar`: Points to the tool currently selected in the toolbar
* `active`: Points to the tool currently used when drawing or interacting with the application. Hence `active` points either to `toolbar` selected tool or two one of the `Button`s

Reply:
Is there a nice `cpp` conform way to document this in the code? I guess in this case it would be very hard to make the variable name itself self descriptive enough.

Reply:
Okay that is a really complex way of doing stuff. Looks like this system grew some features and now ended up with three arms and one and a half leg instead of two of each.
It would be great if we could simplify this. So it's up to you if you want to do this in this PR or a follow-up or leave it for somebody else but I think this desperately needs some simplification. How exactly though would require some looking into the code and its requirements.

Reply:
You're right some simplification would be good for this part of the code. It was quite complex when I started and I might have even added to the complexity with my recent changes.
I would probably still do it outside of this PR to fix the regression first. Even though the Code is complex at least it works right now. While I'll do the other changes I'll already think about how this could be simplified.